diff --git a/node_modules/@dnd-kit/core/dist/core.esm.js b/node_modules/@dnd-kit/core/dist/core.esm.js
index baea98b..e1a0130 100644
--- a/node_modules/@dnd-kit/core/dist/core.esm.js
+++ b/node_modules/@dnd-kit/core/dist/core.esm.js
@@ -2024,6 +2024,10 @@ class AbstractPointerSensor {
         capture: true
       }); // Remove any text selection from the document
 
+      this.documentListeners.add(EventName.Click, preventDefault, {
+        capture: true
+      });
+
       this.removeTextSelection(); // Prevent further text selection while dragging
 
       this.documentListeners.add(EventName.SelectionChange, this.removeTextSelection);
