diff --git a/node_modules/rich-markdown-editor/dist/nodes/Image.js b/node_modules/rich-markdown-editor/dist/nodes/Image.js
index 3e5017e..ce8c6a0 100644
--- a/node_modules/rich-markdown-editor/dist/nodes/Image.js
+++ b/node_modules/rich-markdown-editor/dist/nodes/Image.js
@@ -172,6 +172,9 @@ class Image extends Node_1.default {
                             overlay: {
                                 backgroundColor: theme.background,
                             },
+                            zoomContainer: {
+                                zIndex: 10000
+                            }
                         }, shouldRespectMaxDimension: true })),
                 React.createElement(Caption, { onKeyDown: this.handleKeyDown(props), onBlur: this.handleBlur(props), className: "caption", tabIndex: -1, role: "textbox", contentEditable: true, suppressContentEditableWarning: true, "data-caption": this.options.dictionary.imageCaptionPlaceholder }, alt)));
         };
